src/test/java/pageObjects/BasePage.java

```
@@ -0,0 +1,81 @@
+                        return ((JavascriptExecutor)getDriver()).executeScript("return document.readyState").equals("complete");
+                    }
+                };
+        WebDriverWait wait = new WebDriverWait(getDriver(), 30);
+        wait.until(pageLoadCondition);
+    }
+
+    protected boolean waitForElementVisiblity(By webElement, int timeInSec) {
+        WebElement wb = null;
+        try{
+            WebDriverWait wait = new WebDriverWait(getDriver(), timeInSec);
+            wb = wait.until(ExpectedConditions.presenceOfElementLocated(webElement));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return wb.isEnabled();
+    }
+
+    protected void waitForPageLoaded() {
+        ExpectedCondition<Boolean> expectation = new
+                ExpectedCondition<Boolean>() {
+                    public Boolean apply(WebDriver driver) {
+                        return ((JavascriptExecutor) getDriver()).executeScript("return document.readyState").toString().equals("complete");
+                    }
+                };
+        try {
+            Thread.sleep(1000);
+            WebDriverWait wait = new WebDriverWait(getDriver(), 30);
+            wait.until(expectation);
+        } catch (Throwable error) {
+            Assert.fail("Timeout waiting for Page Load Request to complete.");
+        }
+    }
+
+    protected boolean waitForTextWithRetries(String text, int numRetries, long totalWaitTimeSec) {
+        boolean isTextFound = false;
+        try {
+            for (int i = 0; i < numRetries; i++) {
+                System.out.println("wait for: " + text + " - " + i + " number of times");
+                if (getDriver().getPageSource().contains(text)) {
+                    isTextFound = true;
+                    break;
+                }
+                Thread.sleep((totalWaitTimeSec * 1000) / numRetries);
+            }
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```

Same feedback, we shouldn't swallow exceptions.

---

done